falcon/app.py

```diff
@@ -107,7 +135,13 @@ def process_request(self, req, resp):
                                 the on_* responder.
                         \"\"\"
 
-                    def process_resource(self, req, resp, resource, params):
+                        def process_resource(
+                            self,
+                            req: Request,
+                            resp: Response,
+                            resource: object,
+                            params: dict[str, Any],
+                        ) -> None:
                         \"\"\"Process the request and resource *after* routing.
```

Comment:
Do these docstrings need to be escaped that way?

Reply:
we would need to use `'''` there or as the doc string marker.

maybe we should

Reply:
Ah, no need. These are docstrings inside docstring, makes sense. GitHub's diff view tricked me.

Reply:
I feel like it would be nicer to either use `'''` in the outside docs or here.

Reply:
Ruff might decide to change back :slightly_smiling_face: 
I wouldn't like to use `'''` in the user facing docs. We could do on the outside, but as said, Ruff might be unhappy. Let's just keep it as is.

Reply:
I think it likes it if inside the docs there is `"""`.
Like it will keep `'foo " bar'` as is, not change it to `"foo \" bar"`